internal/node/launcher.sh

```diff
@@ -273,23 +273,15 @@ fi
 
 # Tell the node_patches_script that programs should not escape the execroot
 export BAZEL_PATCH_ROOTS="${EXECROOT}"
-# Set all bazel managed node_modules directories as guarded so no symlinks may
-# escape and no symlinks may enter.
-# We always guard against the root node_modules where 1st party deps go.
-# (e.g., /private/.../execroot/build_bazel_rules_nodejs/node_modules)
-export BAZEL_PATCH_ROOTS="${BAZEL_PATCH_ROOTS},${EXECROOT}/node_modules"
+# Make execroot/external a patch guard so that symlinks can't resolve into it
+export BAZEL_PATCH_GUARDS="${EXECROOT}/external"
```

Comment:
feels like it would be nice to have some prose like a README that we keep up-to-date with the code changes? If/when we want to write a new linker we'll need all the goodies that you currently have in your head.

Reply:
Indeed. I should write a few things down while its fresh. Detailing all of the different scenarios will take some effort. Sandbox vs no-sandbox. Windows vs posix. Runfiles vs execroot. Bazel run vs bazel build vs bazel test. etc...